server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -448,6 +449,13 @@ private CommandExecutor startCommandExecutor(
         final Consumer<CommandExecutor> onTakeLeadership = exec -> {
             if (pluginsForLeaderOnly != null) {
                 logger.info("Starting plugins on the leader replica ..");
+                try {
+                    // Wait until the internal project is initialized.
+                    projectInitializer.whenInitialized().get();
```

Comment:
Isn't this only called for the leader?
Shouldn't we do this for all plugins?

Reply:
Others are explicitly invoked after `projectInitializer.initialize()`.
The process happened in the same thread. 
https://github.com/line/centraldogma/blob/e0439173f3f81bc8b3562483bcd9d133f1e40a49/server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java#L1093-L1101

Reply:
Ah I missed that. 😓

Reply:
`onTakeLeadership` is invoked synchronously which is a deadlock. I changed it to be invoked in `PluginGroupStartStop.doStart()`.